sql/core/src/main/scala/org/apache/spark/sql/execution/OptimizeMetadataOnlyQuery.scala

```diff
@@ -119,6 +119,10 @@ case class OptimizeMetadataOnlyQuery(catalog: SessionCatalog) extends Rule[Logic
       }
     }
 
+    if (normalizedFilters.exists(_.find(_.isInstanceOf[Unevaluable]).isDefined)) {
+      return child
+    }
```

Comment:
Why don't you just filter out subquerties consistently with other normalized filters, by `normalizedFilters.filterNot(SubqueryExpression.hasSubquery)`?